apps/root/src/services/priceService.ts

```diff
@@ -183,4 +191,59 @@ export default class PriceService {
 
     return graphData;
   }
+
+  async getPricesForTokenGraph({ token, span, period }: { token: Token; span: number; period: TimeString }) {
+    const wrappedProtocolToken = getWrappedProtocolToken(token.chainId);
+
+    const adjustedToken =
+      token.address === PROTOCOL_TOKEN_ADDRESS
+        ? wrappedProtocolToken
+        : { ...token, address: getTokenAddressForPriceFetching(token.chainId, token.address) };
+
+    const tokens = [{ chainId: token.chainId, token: adjustedToken.address }];
+    try {
+      const prices = await timeoutPromise(
+        this.sdkService.sdk.priceService.getChart({
+          span,
+          period,
+          tokens,
+          bound: {
+            upTo: Math.floor(Date.now() / 1000),
+          },
+        }),
+        '20s'
+      );
+
+      return prices[token.chainId][adjustedToken.address];
+    } catch (e) {
+      console.error(e);
+    }
+  }
+
+  async getTokenPercentageChanges({ token }: { token: Token }): Promise<TokenPercentageChanges> {
+    const today = Math.floor(Date.now() / 1000);
+    const daySeconds = 60 * 60 * 24;
+    const weekSeconds = daySeconds * 7;
+    const monthSeconds = daySeconds * 30;
+    const yearSeconds = daySeconds * 365;
+
+    const prices = [today, daySeconds, weekSeconds, monthSeconds, yearSeconds]
+      .map((seconds) =>
+        this.getUsdHistoricPrice([token], seconds !== today ? (today - seconds).toString() : undefined).then((price) =>
+          Number(formatUnits(price[token.address], 18))
+        )
+      )
+      .map((promise) => timeoutPromise(promise, '10s').catch(console.error));
+
+    const [currentPrice, dayAgoPrice, weekAgoPrice, monthAgoPrice, yearAgoPrice] = await Promise.all(prices);
```

Comment:
we could just use Promise.allSettled

Reply:
Absolutely! This wasn't available due to @types/node version, but now we can use it :D